internal/rest/resources/control.go

```diff
@@ -75,12 +75,26 @@ func controlPost(state state.State, r *http.Request) response.Response {
 		return joinWithToken(state, r, req)
 	}
 
+	reverter := revert.New()
+	defer reverter.Fail()
+	reverter.Add(func() {
+		reExec, err := resetClusterMember(r.Context(), state, true)
+		if err != nil {
+			return
+		}
+
+		// Re-exec the daemon to clear any remaining state.
+		go reExec()
```

Comment:
The revert in `joinWithToken()` also tries to delete the cluster member from the DB after doing the `go reExec()`. Do we need this step here too? 

```go
// Use `force=1` to ensure the node is fully removed, in case its listener hasn't been set up.
err = client.DeleteClusterMember(context.Background(), req.Name, true)
```

Reply:
In the join case, we also have to clear the state of other nodes.

Reply:
Ok thanks makes sense.